frontend/models/product-list/server.ts

```diff
@@ -139,6 +145,23 @@ export function getProductListType(
    }
 }
 
+async function filterDevicesWithNoProducts(
+   productListChildren: ProductListChild[],
+   isPartsList: boolean
+) {
+   if (!isPartsList) {
+      return productListChildren;
+   }
+   const childDevices = productListChildren.reduce(
+      (acc, child) => (child.deviceTitle ? [...acc, child.deviceTitle] : acc),
+      [] as string[]
+   );
+   const devicesWithProducts = await findDevicesWithProducts(childDevices);
+   return productListChildren.filter(
+      (child) => child.deviceTitle && devicesWithProducts[child.deviceTitle] > 0
```

Comment:
Are there some strapi product lists that have no `deviceTitle`?

Reply:
TS thinks it's nullable. Perhaps we can correct our type if it's never null in practice